backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java

```diff
@@ -775,4 +780,55 @@ IGitMacheteRepositorySnapshot discoverGitMacheteRepository() throws GitMacheteEx
     }
 
   }
+
+  @Getter
+  private static final class GitMacheteNonRootBranchesCreationResult {
```

Comment:
Hmm what about two changes to the naming:
1. skip `GitMachete` prefix, it makes more sense in external APIs but this is never exposed
2. rename to `NonRootCreatedAndSkippedBranches` to better communicate the object's contents?

Reply:
Similarly for `Root`; also this will work better with naming like:

`var downstreamBranches = deriveDownstreamBranches(...)`

Currently a `GitMacheteNonRootBranchesCreationResult` is assigned to `downstreamBranches`, which is a bit awkward :/